### PR TITLE
Refactor launch files

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

Related to #165.

This is a refactor of the launch files in the repository to make them more useful for testing on real robots and to make it easier to add different rosbags, maps, and configurations.

* Rename launch files to better convey their purpose (localization, rviz, rosbag, simulation).
* Minor Python code changes to maintain consistency between launch files.
* Remove `/clock` topic and last 10 seconds from `perfect_odometry` rosbag before it collides with the wall.
* Add `use_composition` parameter to run `lifecycle_manager`, `map_server` and `beluga_amcl` in a single process.
* Rename localization node parameters in launch file.
* Move `config/params.yml` to `params/default.yaml`.
* Update the profiling and benchmarking scripts.
* Update documentation.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

💥 **Breaking change!** This is necessary to better support real robot deployments.

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commmits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga_example/launch/utils/localization_launch.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        ),
+        DeclareLaunchArgument(
+            name='localization_node',
+            default_value='amcl_node',
+            description='Localization node to launch.',
+        ),
+        DeclareLaunchArgument(
+            name='localization_plugin',
+            default_value='beluga_amcl::AmclNode',
+            description='Localization node to plugin to use if composition is enabled.',
```
Comment (glpuga) [inline beluga_example/launch/utils/localization_launch.py:50]:
```suggestion
            description='Localization node plugin to use if composition is enabled.',
```

File: beluga_example/launch/utils/localization_launch.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        ),
+        DeclareLaunchArgument(
+            name='localization_prefix',
+            default_value='',
+            description='Set of commands/arguments to preceed the node command (e.g. "timem --").',
```
Comment (glpuga) [inline beluga_example/launch/utils/localization_launch.py:55]:
```suggestion
            description='Set of commands/arguments to precede the node command (e.g. "timem --").',
```

Comment (glpuga) [reply beluga_example/launch/utils/localization_launch.py:55]: Or maybe 
```suggestion
            description='Set of commands/arguments to prefix the node command with (e.g. "timem --").',
```

File: beluga_benchmark/scripts/profiling/profile_amcl_with_bagfile [hunk 3]
```diff
@@ -34,8 +25,6 @@ if [[ "$KPTR_RESTRICT" -ne "0" || "$PERF_EVENT_PARANOID" -ne "-1" ]]; then
...
 fi
 
-0<&- script -qefc "ros2 launch beluga_example common_nodes_launch.py" /dev/null &
-perf record -e cycles -F 99 -g --call-graph dwarf --\
-    /ws/install/beluga_amcl/lib/beluga_amcl/amcl_node --ros-args --log-level info --ros-args -r __node:=amcl \
-    -p use_sim_time:=True --params-file /ws/install/beluga_example/share/beluga_example/config/params.yaml &
-ros2 bag play /ws/install/beluga_example/share/beluga_example/bags/perfect_odometry --rate 3
+PERF_PREFIX="perf record -e cycles -F 99 -g --call-graph dwarf -o perf.data --"
+
+ros2 launch beluga_example perfect_odometry_launch.py localization_prefix:="${PERF_PREFIX}" playback_rate:=3
```
Comment (glpuga) [inline beluga_benchmark/scripts/profiling/profile_amcl_with_bagfile:30]:
```suggestion
ros2 launch beluga_example perfect_odometry_launch.py localization_prefix:="${PERF_PREFIX}" playback_rate:=1
```

**Review by glpuga — DISMISSED** [review]

File: beluga_benchmark/scripts/profiling/profile_amcl_with_bagfile (hunk 3 above)
Comment (nahueespinosa) [reply beluga_benchmark/scripts/profiling/profile_amcl_with_bagfile]: Fixed in https://github.com/Ekumen-OS/beluga/pull/211/commits/9bf5e843a51ed9871a59bd0e8ed8c38d4c437c2f.

[the remark this answers is not in the document]
Comment (nahueespinosa) [reply beluga_benchmark/scripts/profiling/profile_amcl_with_bagfile]: Fixed in https://github.com/Ekumen-OS/beluga/pull/211/commits/9bf5e843a51ed9871a59bd0e8ed8c38d4c437c2f.

File: beluga_example/launch/utils/localization_launch.py (hunk 1 above)
Comment (nahueespinosa) [reply beluga_example/launch/utils/localization_launch.py]: Fixed in https://github.com/Ekumen-OS/beluga/pull/211/commits/6e66c09ad546284f94dcecc207b950c221b52347.

File: beluga_example/launch/utils/localization_launch.py (hunk 2 above)
Comment (nahueespinosa) [reply beluga_example/launch/utils/localization_launch.py]: Fixed in https://github.com/Ekumen-OS/beluga/pull/211/commits/6e66c09ad546284f94dcecc207b950c221b52347.

File: beluga_example/README.md [hunk 4]
```diff
@@ -6,17 +6,22 @@ This package contains example launch files that demonstrate the use of Beluga-ba
...
   ```
 
-- Launch a simulation with Beluga AMCL as a composable node.
+- Launch a simulation that can be teleoperated together with Beluga AMCL.
   ```bash
-  ros2 launch beluga_example example_component_launch.py
+  ros2 launch beluga_example simulation.launch.xml
+  ```
+
+- Launch Beluga AMCL, a map server and a lifecycle manager, useful for testing on real robots.
```
Comment (hidmic) [inline beluga_example/README.md:24]: @nahueespinosa nit^2: 
```suggestion
- Launch Beluga AMCL, a map server, and a lifecycle manager. Useful for testing on real robots.
```

**Review by hidmic — APPROVED** [review]
LGTM

File: beluga_example/README.md (hunk 4 above)
Comment (nahueespinosa) [reply beluga_example/README.md]: Done in https://github.com/Ekumen-OS/beluga/pull/211/commits/2dcf64926eab12eaa968be560ba444e00700692b.

**Comment by nahueespinosa** [comment]
@hidmic @glpuga Thanks for the reviews! Pushing new commits dismisses previous approvals, can I get a final LGTM?

**Review by hidmic — APPROVED** [review]

**Pull request merged**